### PR TITLE
Add semester and module view

### LINKED ISSUE #65: Where should semester view and module view lead to?
Semester view: show all courses offered in that semester

Modules view: show all courses of that module sorted by semester 

Implementation: reuse ‘courses’ view

### PR DESCRIPTION
**Pull request by Na-Nazhou**
Closes #65

### CODE REVIEW

File: controllers/courses_controller.js [hunk 1]
```diff
@@ -3,14 +3,35 @@ const sql = require('../sql');
...
+        res.render('courses', { data: data.rows, title: `Courses offered in ${query.semester_name}` });
+      }
+    });
+  } else if (query.module_code) {
+    db.query(sql.courses.queries.get_courses_by_module, [query.module_code], (err, data) => {
+      if (err) {
+        log.error(`Failed to get courses for ${query.module_code}`);
+        next(err);
+      } else {
+        res.render('courses', { data: data.rows, title: `Courses: ${query.module_code}` });
```
Comment (gary-lgy) [inline controllers/courses_controller.js]:
```suggestion
        res.render('courses', { data: data.rows, title: `Courses for module ${query.module_code}` });
```

File: controllers/courses_controller.js [hunk 2]
```diff
@@ -3,14 +3,35 @@ const sql = require('../sql');
...
 exports.index = (req, res, next) => {
-  db.query(sql.courses.queries.get_courses, (err, data) => {
-    if (err) {
-      log.error('Failed to get courses');
-      next(err);
-    } else {
-      res.render('courses', { data: data.rows });
-    }
-  });
+  const { query } = req;
```
Comment (gary-lgy) [inline controllers/courses_controller.js]: Destructure here to obtain both `semester_name` and `module_code` then proceed to check if they are `undefined`

File: views/modules.ejs [hunk 3]
```diff
@@ -21,7 +21,7 @@
       <tr>
         <td><%= data[i].module_code%></td>
         <td>
-          <a href="#" class="btn btn-primary">
+          <a href="/courses?module_code=<%= encodeURIComponent(data[i].module_code)%>" class="btn btn-primary">
```
Comment (gary-lgy) [inline views/modules.ejs]: Define a route helper for this. Same for `views/semesters.ejs`

**Review by gary-lgy — CHANGES_REQUESTED** [review]

File: views/modules.ejs (hunk 3 above)
Comment (Na-Nazhou) [reply views/modules.ejs]: How should i name the helper?

**Pull request merged**